tempesta_fw/str.c

```diff
@@ -36,14 +36,14 @@ tfw_str_del_chunk(TfwStr *str, int id)
 
 	if (TFW_STR_CHUNKN(str) == 2) {
 		/* Just fall back to plain string. */
-		*str = *((TfwStr *)str->ptr + (id ^ 1));
+		*str = *((TfwStr *)str->chunks + (id ^ 1));
```

Comment:
We introduced `TfwStr->chunks` to avoid type conversions. Please remove **all** unnecessary type conversions.